iceoryx_hoofs/legacy/include/iceoryx_hoofs/cxx/functional_interface.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_CXX_FUNCTIONAL_INTERFACE_HPP
+#define IOX_HOOFS_CXX_FUNCTIONAL_INTERFACE_HPP
+
+#include "iox/functional_interface.hpp"
+
+namespace iox
+{
+/// @todo iox-#1593 Deprecate include
+/// [[deprecated("Deprecated in 3.0, removed in 4.0, please include 'iox/functional_interface.hpp' instead")]]
+namespace cxx
+{
+namespace internal
+{
+/// @deprecated use `iox::internal::HasValueMethod` instead of `iox::cxx::internal::HasValueMethod`
+using iox::internal::HasValueMethod;
```

Comment:
@FerdinandSpitzschnueffler Should be no problem here but see https://stackoverflow.com/questions/6175705/scope-of-using-declaration-within-a-namespace

I am not sure it can lead to problems further down the road but it is deprecated. We can merge it as is but I have to think about this in more detail...

Reply:
@FerdinandSpitzschnueffler 

It should be OK because:
1) There is no ambiguity as the function in question should not be in `iox::cxx::internal` (because it was moved and the namespace is initially empty if we ignore the using directives).
2) All required declarations are available by including `"iox/functional_interface.hpp"`.

This must hold by design and should be necessary and sufficient.

Reply:
Agree. Thanks for your research!